hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/GeneralThrottleAccumulator.java

```diff
@@ -106,6 +109,44 @@ public boolean shouldThrottle(@NonNull TransactionInfo txnInfo, Instant t, Heder
         return false;
     }
 
+    public boolean shouldThrottleQuery(HederaFunctionality queryFunction, Instant now, Query query) {
+        final var configuration = configProvider.getConfiguration();
+        final var shouldThrottleByGas =
+                configuration.getConfigData(ContractsConfig.class).throttleThrottleByGas();
+
+        // Note that by payer exempt from throttling we mean just that those transactions will not be throttled,
+        // such payer accounts neither impact the throttles nor are they impacted by them
+        // In the current mono-service implementation we have the same behavior, additionally it is
+        // possible that transaction can also be exempt from affecting congestion levels separate from throttle
+        // exemption
+        // but this is only possible for the case of triggered transactions which is not yet implemented (see
+        // MonoMultiplierSources.java)
+        final var payer = PbjConverter.toPbj(query.getAccountDetails().getAccountId());
```

Comment:
Is this a correct way to get the payer?

Reply:
We are in the modularized code-base and should not use old protobuf objects in the method signatures. We should use PBJ-objects as long as possible and only convert right before and after we call mono-code.

The parameter `query` has to have the full name `com.hedera.hapi.node.transaction.Query`. When you do that, `query.accountDetails().account()` will have the right type.